### PR TITLE
Fix typos, scaladoc tags, and some minor code smells.

### PR DESCRIPTION
**Pull request by retronym**

### CODE REVIEW

**Comment by retronym** [comment]
Review by @DarkDimius

**Comment by DarkDimius** [comment]
@retronym thanks, where you running some kind of tool? Or this is found by reading code.

File: src/dotty/tools/dotc/Run.scala [hunk 1]
```diff
@@ -37,10 +37,10 @@ class Run(comp: Compiler)(implicit ctx: Context) {
       throw ex
   }
 
-  /** TODO: There's a fundamental design problem here: We assmble phases using `squash`
+  /** TODO: There's a fundamental design problem here: We assemble phases using `squash`
```
Comment (DarkDimius) [inline src/dotty/tools/dotc/Run.scala]: this comment is out-of-date. We already assemble phases on each run.

**Comment by retronym** [comment]
@DarkDimius I used IntelliJ's inspections / quick fixes.

**Pull request merged**